service/s3/internal/customizations/update_endpoint_test.go

```diff
@@ -134,9 +135,9 @@ func Test_UpdateEndpointBuild(t *testing.T) {
 				},
 				tests: []s3BucketTest{
 					{"abc", "key", "https://example.region.amazonaws.com/abc/key?x-id=GetObject", ""},
-					{"a$b$c", "key", "https://example.region.amazonaws.com/a%24b%24c/key?x-id=GetObject", ""},
-					{"a.b.c", "key", "https://example.region.amazonaws.com/a.b.c/key?x-id=GetObject", ""},
-					{"a..bc", "key", "https://example.region.amazonaws.com/a..bc/key?x-id=GetObject", ""},
+					{"a$b$c", "key", "", "cannot be used with"},
```

Comment:
this is breaking, no? why are these use cases no longer supported?

Reply:
It's the "it never worked anyway and now it's just explicitly caught instead of producing garbage" case. I mentioned this in a standup a week or so back.

Reply:
I believe in this case it was accelerate+path style now being caught correctly (it's one of those endpoint combinations).